src/adcs_handler.c

```diff
@@ -46,242 +50,450 @@ ADCS_returnState adcs_telecommand(uint32_t * command, uint32_t length) {
+//  uint32_t command = RESET_UPLOAD_BLOCK_ID;
+//  return adcs_telecommand(&command, 1);
+//}
+//
+///**
+// * @brief
+// * 		Reset File List Read Pointer
+// * @return
+// * 		<if applicable, brief description of what is returned by the
+// function>
+// */
+// ADCS_returnState ADCS_reset_file_list_read_pointer(){
+//  uint32_t command = RESET_FILE_LIST_READ_POINTER_ID;
+//  return adcs_telecommand(&command, 1);
+//}
+//
+///**
+// * @brief
+// * 		Initiate Download Burst
+// * @param msg_length
+// * 		Message Length
+// * @param ignore_hole_map
+// * 		Ignore Hole Map
+// * @return
+// * 		<if applicable, brief description of what is returned by the
+// function>
+// */
+// ADCS_returnState ADCS_initiate_download_burst(uint8_t msg_length, bool
+// ignore_hole_map){
+//  uint32_t command[3];
+//  command[0] = INITIATE_DOWNLOAD_BURST_ID;
+//  command[1] = msg_length;
+//  command[2] = ignore_hole_map;
+//  adcs_telecommand(command, 3);
+//}
 
+/*************************** BootLoader TCs ***************************/
 /**
  * @brief
- * 		Reset Boot Registers
- * @details
- * 		<more detailed description of the function>
+ * 		Clear error flags
  * @return
- * 		<if applicable, brief description of what is returned by the function>
+ * 		Success of function defined in adcs_types.h
  */
-ADCS_returnState ADCS_reset_boot_registers() {
-  uint32_t command = RESET_BOOT_REGISTERS_ID;
+ADCS_returnState ADCS_clear_err_flags(void) {
+  uint8_t command = CLEAR_ERR_FLAGS_ID;
   return adcs_telecommand(&command, 1);
 }
 
 /**
  * @brief
- * 		Format SD Card
- * @details
- * 		<more detailed description of the function>
- * @return
- * 		<if applicable, brief description of what is returned by the function>
- */
-ADCS_returnState ADCS_format_sd_card() {
-  uint32_t command[2];
-  command[0] = FORMAT_SD_CARD_ID;
-  command[1] = ADCS_MAGIC_NUMBER; // magic number
-  return adcs_telecommand(command, 2);
-}
-
-/**
- * @brief
- * 		Erase file
- * @details
- * 		<more detailed description of the function>
- * @param file_type
- * 		Accepted parameters: FILE_TYPE_TELEMETERY_LOG, FILE_TYPE_JPG_IMAGE, 
- *                          FILE_TYPE_BMP_IMAGE or FILE_TYPE_INDEX.
- * @param file_counter
- * 		<if applicable, brief (one line) description of one of the function's arguments>
- * @param erase_all
- * 		<if applicable, brief (one line) description of one of the function's arguments>
+ * 		Set boot index
+ * @param index
+ * 		one valid value: 1 = internal flash program
  * @return
- * 		<if applicable, brief description of what is returned by the function>
+ * 		Success of function defined in adcs_types.h
  */
-ADCS_returnState ADCS_erase_file(File_Type file_type, uint8_t file_counter, bool erase_all) {
-  uint32_t command[4];
-  command[0] = ERASE_FILE_ID;
-  command[1] = file_type;
-  command[2] = file_counter;
-  command[3] = erase_all;
-  return adcs_telecommand(command, 4);
+ADCS_returnState ADCS_set_boot_index(uint8_t index) {
```

Comment:
Didn't make sense to me why `index` is an enum in the manual, since it's always 1. Is 'Internal Flash Program' the only program that can be selected to run? The program values list for `ADCS_read_program_info` and `ADCS_copy_program_internal_flash` commands has more programs listed so it makes more sense for that index to be an enum.

Reply:
Agreed. Yes.